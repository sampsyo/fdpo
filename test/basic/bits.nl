# ARGS: x=3 y=7
in x: 8;
in y: 8;
out a: 8;
out o: 8;
out xo: 8;
a = and[8](x, y);
o = or[8](x, y);
xo = xor[8](x, y);
