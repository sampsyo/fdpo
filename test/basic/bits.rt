in x: 8;
in y: 8;
out a: 8;
out o: 8;
out xo: 8;
out left: 8;
out right: 8;
out logic: 8;
out arith: 8;
a = and[8](x, y);
o = or[8](x, y);
xo = xor[8](x, y);
left = shl[8](x, 8d1);
right = shr[8](x, 8d1);
logic = shr[8](8b10000000, 8d1);
arith = ashr[8](8b10000000, 8d1);
