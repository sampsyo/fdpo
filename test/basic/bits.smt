(let ((.def_0 (bvashr #b10000000 #b00000001))) (let ((.def_1 (= arith .def_0))) (let ((.def_2 (bvlshr #b10000000 #b00000001))) (let ((.def_3 (= logic .def_2))) (let ((.def_4 (bvlshr x #b00000001))) (let ((.def_5 (= right .def_4))) (let ((.def_6 (bvshl x #b00000001))) (let ((.def_7 (= left .def_6))) (let ((.def_8 (bvxor x y))) (let ((.def_9 (= xo .def_8))) (let ((.def_10 (bvor x y))) (let ((.def_11 (= o .def_10))) (let ((.def_12 (bvand x y))) (let ((.def_13 (= a .def_12))) (let ((.def_14 (and .def_13 .def_11 .def_9 .def_7 .def_5 .def_3 .def_1))) .def_14)))))))))))))))
