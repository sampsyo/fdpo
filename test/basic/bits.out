a = 3
o = 7
xo = 4
left = 6
right = 1
logic = 64
arith = 192
